webSocket/server/server.go

```diff
@@ -152,51 +169,21 @@ func (s *server) SetPayloadHandler(handler webSocket.PayloadHandler) error {
 	return nil
 }
 
-// Listen will switch the server in listen mode and the server will start to listen from messages from the new connections
-func (s *server) Listen() {
-	// TODO refactor this method
-	s.connectionHandler = func(connection webSocket.WSConClient) {
-		webSocketsReceiver, err := receiver.NewReceiver(receiver.ArgsReceiver{
-			PayloadConverter:   s.payloadConverter,
-			Log:                s.log,
-			RetryDurationInSec: int(s.retryDuration.Seconds()),
-			BlockingAckOnError: s.blockingAckOnError,
-		})
-		if err != nil {
-			s.log.Warn("s.connectionHandler cannot create receiver", "error", err)
-		}
-		err = webSocketsReceiver.SetPayloadHandler(s.payloadHandler)
-		if err != nil {
-			s.log.Warn("s.SetPayloadHandler cannot set payload handler", "error", err)
-		}
-
-		go func() {
-			s.receivers.AddReceiver(connection.GetID(), webSocketsReceiver)
-			// this method is blocking
-			_ = webSocketsReceiver.Listen(connection)
-			// if method listen will end, the client was disconnected, and we should remove the listener from the list
-			s.receivers.RemoveReceiver(connection.GetID())
-		}()
-	}
-
-}
-
 // Close will close the server
 func (s *server) Close() error {
 	err := s.httpServer.Shutdown(context.Background())
 	if err != nil {
 		s.log.Warn("server.Close() cannot close http server", "error", err)
 	}
 
-	err = s.sender.Close()
-	if err != nil {
-		s.log.Warn("server.Close() cannot close the sender", "error", err)
-	}
-
-	for _, webSocketsReceiver := range s.receivers.GetAll() {
-		err = webSocketsReceiver.Close()
+	for _, tuple := range s.transceiversAndConn.GetAll() {
+		err = tuple.Transceiver.Close()
+		if err != nil {
+			s.log.Warn("server.Close() cannot close transceiver", "error", err)
+		}
+		err = tuple.Conn.Close()
 		if err != nil {
-			s.log.Warn("server.Close() cannot close receiver", "error", err)
+			s.log.Warn("server.Close() cannot close connection", "id", tuple.Conn.GetID(), "error", err.Error())
 		}
 	}
 
```

Comment:
so we never return error in case the shutdown method failed or a transceiver close method failed? Is this ok?